src/init.c

```diff
@@ -618,7 +618,28 @@ static int hyper_kill_container(char *json, int length)
 		goto out;
 	}
 
-	kill(c->exec.pid, (int)json_object_get_number(json_object(value), "signal"));
+	const int all_processes = json_object_get_boolean(json_object(value),
+		"allProcesses");
+
+	const int signal = (int)json_object_get_number(json_object(value), "signal");
+
+	struct hyper_exec *exec;
+	/* json_object_get_boolean returns '1' only when
+	 * 'allProcesses' exists and is true
+	 */
+	if (1 == all_processes) {
```

Comment:
@devimc the 'const == var' idiom is a valid one (but can cause almost as much debate 'as which editor do you use?'), but as @amshinde notes, it does not seem to be the defacto style in this file, so in this case I think we should stick with 'var == 1'...